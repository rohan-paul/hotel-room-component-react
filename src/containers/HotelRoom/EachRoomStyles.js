import React from "react"
import { makeStyles } from "@material-ui/core/styles"

const useStyles = makeStyles(theme => ({
  container: {
    margin: "auto",
    backgroundColor: theme.common.white,
  },

  tableContainer: {
    position: "relative",
    margin: "auto",
    maxWidth: "1100px",
    height: "80px",
    // minHeight: "calc(100vh - 100px)",
    alignItems: "center ",
    justifyContent: "center",
    verticalAlign: "middle",
    textAlign: "center",
    border: "1px solid #aaa",
  },
  buttonDisabled: {
    color: theme.palette.primary.dark,
  },

  avatar: {
    width: "30px",
    height: "30px",
  },
  roomsContainer: {
    height: "70px",
    float: "left",
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    textAlign: "center",
    // backgroundColor: "red",
  },
  ctaContainer: {
    height: "70px",
    float: "right",
    marginRight: "15px",
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    textAlign: "center",
    // backgroundColor: "green",
  },
  leftTextContainer: {
    verticalAlign: "middle",
    alignItems: "center",
    textAlign: "center",
  },
}))

export { useStyles }
