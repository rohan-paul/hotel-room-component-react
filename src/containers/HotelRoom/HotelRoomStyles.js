import React from "react"
import { makeStyles } from "@material-ui/core/styles"

const useStyles = makeStyles(theme => ({
  container: {
    margin: "auto",
    backgroundColor: theme.common.white,
  },
}))

export { useStyles }
