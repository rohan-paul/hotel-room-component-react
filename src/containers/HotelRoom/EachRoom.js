import React from "react"
import { useSelector, useDispatch } from "react-redux"
import { Typography } from "@material-ui/core"
import IconButton from "@material-ui/core/IconButton"
import { useStyles } from "./EachRoomStyles"

const EachRoom = ({
  rooms,
  adults,
  children,
  leftText,
  incrementRoomsNo,
  decrementRoomsNo,
  incrementAdultsNo,
  decrementAdultsNo,
  incrementChildrenNo,
  decrementChildrenNo,
}) => {
  const globalStore = useSelector(state => state.globalStore)
  const classes = useStyles()
  const dispatch = useDispatch()
  const { no_of_rooms, no_of_adults, no_of_children } = globalStore

  const handleIncrementClick = () => {
    if (rooms) {
      dispatch(incrementRoomsNo())
    } else if (adults) {
      dispatch(incrementAdultsNo())
    } else if (children) {
      dispatch(incrementChildrenNo())
    }
  }

  const handleDecrementClick = () => {
    if (rooms) {
      dispatch(decrementRoomsNo())
    } else if (adults) {
      dispatch(decrementAdultsNo())
    } else if (children) {
      dispatch(decrementChildrenNo())
    }
  }

  const maxPersons = no_of_rooms * 4

  return (
    <>
      <div className={classes.tableContainer}>
        <div className={classes.roomsContainer}>
          <IconButton
            aria-label="Filter list"
            classes={{ root: classes.button, disabled: classes.buttonDisabled }}
          >
            <img
              src={
                adults
                  ? require("../../assets/images/adult-icon.png")
                  : children
                  ? require("../../assets/images/child-icon.png")
                  : rooms
                  ? require("../../assets/images/bed-icon.png")
                  : null
              }
              className={classes.avatar}
              alt=""
            />
          </IconButton>
          <div className={classes.leftTextContainer}>
            <Typography variant="subtitle1">{leftText}</Typography>
          </div>
        </div>
        <div className={classes.ctaContainer}>
          <IconButton
            aria-label="Filter list"
            className={classes.adultIconContainer}
            disabled={
              (rooms && no_of_rooms <= 1) ||
              (adults && no_of_adults <= 1) ||
              (children && no_of_children <= 0)
            }
            onClick={handleDecrementClick}
          >
            <img
              src={
                (rooms && no_of_rooms <= 1) ||
                (adults && no_of_adults <= 1) ||
                (children && no_of_children <= 0)
                  ? require("../../assets/images/minus-icon-disabled.png")
                  : require("../../assets/images/minus-icon.png")
              }
              className={classes.avatar}
              alt=""
            />
          </IconButton>
          <div className={classes.leftTextContainer}>
            <Typography variant="subtitle1">
              {rooms
                ? no_of_rooms
                : adults
                ? no_of_adults
                : children
                ? no_of_children
                : null}
            </Typography>
          </div>
          <IconButton
            aria-label="Filter list"
            className={classes.adultIconContainer}
            onClick={handleIncrementClick}
            disabled={
              (rooms && no_of_rooms >= 5) ||
              (adults && no_of_adults + no_of_children >= maxPersons) ||
              (children && no_of_adults + no_of_children >= maxPersons)
            }
          >
            <img
              src={
                (rooms && no_of_rooms >= 5) ||
                (adults && no_of_adults + no_of_children >= maxPersons) ||
                (children && no_of_adults + no_of_children >= maxPersons)
                  ? require("../../assets/images/plus-icon-disabled.png")
                  : require("../../assets/images/plus-icon.png")
              }
              className={classes.avatar}
              alt=""
            />
          </IconButton>
        </div>
      </div>
    </>
  )
}

export default EachRoom
