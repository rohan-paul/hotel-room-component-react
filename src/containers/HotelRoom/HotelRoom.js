/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/button-has-type */
/* eslint-disable-next-line react-hooks/exhaustive-deps */
import React from "react"
// import PropTypes from "prop-types"
// import { useSelector, useDispatch } from "react-redux"
import { useStyles } from "./HotelRoomStyles"
import EachRoom from "./EachRoom"
import {
  incrementAdultsNo,
  decrementAdultsNo,
  incrementRoomsNo,
  decrementRoomsNo,
  incrementChildrenNo,
  decrementChildrenNo,
} from "../../actions/getUserActions"

const HotelRoom = () => {
  const classes = useStyles()
  // const globalStore = useSelector(state => state.globalStore)
  // const dispatch = useDispatch()

  return (
    <div className={classes.container}>
      <EachRoom
        rooms
        leftText={"ROOMS"}
        incrementRoomsNo={incrementRoomsNo}
        decrementRoomsNo={decrementRoomsNo}
      ></EachRoom>
      <EachRoom
        adults
        leftText={"ADULTS"}
        incrementAdultsNo={incrementAdultsNo}
        decrementAdultsNo={decrementAdultsNo}
      ></EachRoom>
      <EachRoom
        children
        leftText={"CHILDREN"}
        incrementChildrenNo={incrementChildrenNo}
        decrementChildrenNo={decrementChildrenNo}
      ></EachRoom>
    </div>
  )
}

// HotelRoom.propTypes = {
//   globalStore: PropTypes.object,
// }

export default HotelRoom
