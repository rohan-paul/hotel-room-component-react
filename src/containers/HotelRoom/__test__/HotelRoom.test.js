import React from "react"
import renderer from "react-test-renderer"
import { MuiThemeProvider } from "@material-ui/core/styles"
import globalTheme from "../../../globalTheme"
import PropTypes from "prop-types"
import { mount, shallow } from "enzyme"
import expect from "expect"
import { createShallow, createMount } from "@material-ui/core/test-utils"
import configureMockStore from "redux-mock-store"
import thunk from "redux-thunk"
import { Provider } from "react-redux"
import HotelRoom from "../HotelRoom"
import EachRoom from "../EachRoom"

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

const store = mockStore({
  globalStore: {
    no_of_rooms: 1,
    no_of_adults: 2,
    no_of_children: 0,
  },
})

describe("GithubMostPopularList Component", () => {
  it("should match snapshot", () => {
    const matches = (children, theme = globalTheme) =>
      expect(
        renderer
          .create(
            <MuiThemeProvider theme={globalTheme}>{children}</MuiThemeProvider>,
          )
          .toJSON(),
      ).toMatchSnapshot()
  })

  it("should render a startup component if startup is not complete", () => {
    const wrapper = mount(
      <Provider store={store}>
        <MuiThemeProvider theme={globalTheme}>
          <HotelRoom />
        </MuiThemeProvider>
      </Provider>,
    )
    expect(wrapper.find("EachRoom").length).toEqual(3)
  })
})
