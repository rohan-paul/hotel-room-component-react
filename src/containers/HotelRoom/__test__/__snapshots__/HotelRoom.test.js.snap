// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelRoom Component should match snapshot 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="makeStyles-tableContainer-3"
  >
    <div
      className="makeStyles-roomsContainer-6"
    >
      <button
        aria-label="Filter list"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Click Me"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt=""
            className="makeStyles-avatar-5"
            src="bed-icon.png"
          />
        </span>
      </button>
      <div
        className="makeStyles-leftTextContainer-8"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          ROOMS
        </h6>
      </div>
    </div>
    <div
      className="makeStyles-ctaContainer-7"
    >
      <button
        aria-label="Filter list"
        className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt=""
            className="makeStyles-avatar-5"
            src="minus-icon-disabled.png"
          />
        </span>
      </button>
      <div
        className="makeStyles-leftTextContainer-8"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          1
        </h6>
      </div>
      <button
        aria-label="Filter list"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt=""
            className="makeStyles-avatar-5"
            src="plus-icon.png"
          />
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-tableContainer-3"
  >
    <div
      className="makeStyles-roomsContainer-6"
    >
      <button
        aria-label="Filter list"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Click Me"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt=""
            className="makeStyles-avatar-5"
            src="adult-icon.png"
          />
        </span>
      </button>
      <div
        className="makeStyles-leftTextContainer-8"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          ADULTS
        </h6>
      </div>
    </div>
    <div
      className="makeStyles-ctaContainer-7"
    >
      <button
        aria-label="Filter list"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt=""
            className="makeStyles-avatar-5"
            src="minus-icon.png"
          />
        </span>
      </button>
      <div
        className="makeStyles-leftTextContainer-8"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          2
        </h6>
      </div>
      <button
        aria-label="Filter list"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt=""
            className="makeStyles-avatar-5"
            src="plus-icon.png"
          />
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-tableContainer-3"
  >
    <div
      className="makeStyles-roomsContainer-6"
    >
      <button
        aria-label="Filter list"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Click Me"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt=""
            className="makeStyles-avatar-5"
            src="child-icon.png"
          />
        </span>
      </button>
      <div
        className="makeStyles-leftTextContainer-8"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          CHILDREN
        </h6>
      </div>
    </div>
    <div
      className="makeStyles-ctaContainer-7"
    >
      <button
        aria-label="Filter list"
        className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt=""
            className="makeStyles-avatar-5"
            src="minus-icon-disabled.png"
          />
        </span>
      </button>
      <div
        className="makeStyles-leftTextContainer-8"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          0
        </h6>
      </div>
      <button
        aria-label="Filter list"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt=""
            className="makeStyles-avatar-5"
            src="plus-icon.png"
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;
